fsrs/src/main/kotlin/com/inout/fsrs/schedule/SchedulingCard.kt

```diff
@@ -55,19 +57,32 @@ class SchedulingCard(card: Card, now: Date) {
         return this
     }
 
-    fun schedule(now: Date, hardInterval: Int, goodInterval: Int, easyInterval: Int): SchedulingCard {
+    fun schedule(
+        now: Instant,
+        hardInterval: Int,
+        goodInterval: Int,
+        easyInterval: Int,
+    ): SchedulingCard {
         again.scheduledDays = 0
         hard.scheduledDays = hardInterval
         good.scheduledDays = goodInterval
         easy.scheduledDays = easyInterval
-        again.due = now.addMinutes(5)
-        hard.due = if (hardInterval > 0) now.addDays(hardInterval) else now.addMinutes(10)
-        good.due = now.addDays(goodInterval)
-        easy.due = now.addDays(easyInterval)
+        again.due = now.plusMinutes(5)
+        hard.due =
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/in-out-team/api-server/pull/11#issuecomment-2362318726>#0b18f5</a><span> - 09/20/2024, 12:05 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The when expression for setting 'hard.due' doesn't handle negative 'hardInterval' values explicitly, which could lead to unexpected behavior.
</div>
<div id="fix">
<b>Fix:</b> Modify the when expression to explicitly handle negative hardInterval values. Consider throwing an exception or using a default value for negative intervals.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -71,6 +71,7 @@ class SchedulingCard(card: Card, now: Instant) {
      hard.due =
          when {
              hardInterval > 0 -> now.plusDays(hardInterval)
 +            hardInterval < 0 -> throw IllegalArgumentException("hardInterval must be non-negative")
              else -> now.plusMinutes(10)
          }
      good.due = now.plusDays(goodInterval)
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged